### PR TITLE
Rework context

### PR DESCRIPTION
**Pull request by pmcelhaney**
A minor breaking change in the way context objects are defined.

## The file is now named `_.context.ts` instead of `$.context.ts`.

This change was made because putting a `$` in a file name can be confusing in Mac/Linux/Unix systems.

### Old

```ts
class Context {
  // ...
}

export default new Context();
```

### New

```ts
export class Context {
  // ...
}
```

In the future, the context class will be able to have a constructor that takes one argument, an instance of the _parent_ context.

```ts
// paths/accounts/_context.ts
import { CounterfactContext } from "../../types.ts";

export class Context extends CounterfactContext {
  constructor(registry) {
    this.mainContext = registry.find("/");
    this.productsContext = registry.find("/products");
  }
  // ...
}
```

### CODE REVIEW

**Review by dethell — DISMISSED** [review]

File: docs/usage.md [hunk 1]
```diff
@@ -140,7 +140,7 @@ export const POST: HTTP_POST = ($) => {
  };
 ```
 
-The `context` object is defined in `$.context.ts` in the same directory as the file that uses it. It's up to you to define the API for a context object. For example, your `$.context.ts` file might look like this.
+The `context` object is defined in `_.context.ts` in the same directory as the file that uses it. It's up to you to define the API for a context object. For example, your `_.context.ts` file might look like this.
 
 ```ts
 class PetStore {
```
Comment (dethell) [inline docs/usage.md]:
```suggestion
export class Context {
```

File: docs/usage.md [hunk 2]
```diff
@@ -160,7 +160,7 @@ class PetStore {
 export default new PetStore();
```
Comment (dethell) [inline docs/usage.md]:
```suggestion
```

**Review by dethell — CHANGES_REQUESTED** [review]
Nearly there. Couple changes.

**Comment by pmcelhaney** [comment]
Thanks @dethell! So nice to have another pair of eyes on this stuff.

File: src/typescript-generator/generate.js [hunk 3]
```diff
@@ -86,7 +85,7 @@ export async function generate(
     });
   });
 
-  repository.get("paths/$.context.ts").exportDefault(new ContextCoder(paths));
+  //   repository.get("paths/$.context.ts").exportDefault(new ContextCoder(paths));
```
Comment (dethell) [inline src/typescript-generator/generate.js:88]:
```suggestion
```
Did you mean to leave this comment?

**Review by dethell — APPROVED** [review]
Approved with a minor nit.

**Pull request merged**